docs/content/en/docs/yaml-crd-ref/taskdefinition.md

```diff
@@ -3,3 +3,258 @@ title: KeptnTaskDefinition
 description: Define tasks that can be run pre- or post-deployment
 weight: 89
 ---
+
+
+A `KeptnTaskDefinition` defines tasks
+that are run by the Keptn Lifecycle Toolkit
+as part of the pre- and post-deployment phases of a
+[KeptnApp](./app.md) or
+[workload](../concepts/workloads/).
+
+## Yaml Synopsis
+
+```yaml
+apiVersion: lifecycle.keptn.sh/v?alpha?
+kind: KeptnTaskDefinition
+metadata:
+  name: <task-name>
+spec:
+  function:
+    inline | httpRef | functionRef:
+    [parameters:
+      map:
+        textMessage: "This is my configuration"]
+    [secureParameters:
+      secret: slack-token]
```

Comment:
Let's not use these linux optional brackets as it is used in linux man pages. It's not very common in kubernetes docs
```suggestion
    parameters:
      map:
        textMessage: "This is my configuration"
    secureParameters:
      secret: slack-token
```